app/scripts/controllers/accounts.js

```diff
@@ -0,0 +1,245 @@
+          this.pluginAccounts.remove(fromDomain, address)
+        }
+      })
+    }
+
+    const currentState = this.store.getState()
+    const accountrings = currentState.accountrings
+    const isAccountRingsAccount = accountrings.find(accountring => accountring.accounts.includes(address))
+    if (isAccountRingsAccount) {
+      let newAccountRings = []
+      accountrings.forEach(accountring => {
+        const newAccountRingsAccounts = accountring.accounts.filter(account => account !== address)
+        if (newAccountRingsAccounts.length) {
+          newAccountRings.push({ ...accountring, accounts: newAccountRingsAccounts })
+        }
+      })
+      this.store.updateState({
+        ...currentState,
+        accountrings: newAccountRings,
+      })
+    }
+
+    try {
+      const removeAccountResult = await this.keyringController.removeAccount(address)
+      return removeAccountResult
+    } catch (e) {
+      if (e.message === 'No keyring found for the requested account.' && isAccountRingsAccount) {
+        return address
+      } else {
+        throw e
+      }
+    }
+  }
+
+  async fullUpdate () {
+    const update = await this.keyringController.fullUpdate()
+    const pluginAccounts = this.pluginAccounts.resources
+
+    const pluginTypes = pluginAccounts.map((account) => {
+      return account.fromDomain
+    })
+      .filter(onlyUnique)
+    update.keyringTypes = update.keyringTypes.concat(pluginTypes)
+
+    const pluginKeyrings = pluginTypes.map((domain) => {
+      const accounts = pluginAccounts.filter((account) => {
+        return account.fromDomain === domain
+      })
+        .map(account => normalizeAddress(account.address))
+
+      return {
+        type: domain,
+        accounts,
+      }
+    })
+
+    update.accountrings = update.keyrings.concat(pluginKeyrings)
+    this.store.updateState(update)
+    return update
+  }
+
+  _getUniquePluginAccountDomains (address) {
+    const pluginAccounts = this.pluginAccounts.resources
+    const accounts = pluginAccounts.filter(acct => normalizeAddress(acct.address) === address)
+    const domains = accounts.map(acct => acct.fromDomain)
+    const uniquePluginAccountDomains = domains.filter(onlyUnique)
+    return uniquePluginAccountDomains
+  }
+
+  getHandlerForAccount (address) {
+    const uniquePluginAccountDomains = this._getUniquePluginAccountDomains(address)
+
+    if (uniquePluginAccountDomains.length > 1) {
+      throw new Error(`Multiple plugins claiming ownership of account ${address}, please request from plugin directly.`)
+    }
+
+    if (!this.pluginsController) {
+      throw new Error('No handlers exist to manage account.')
+    }
+
+    const handler = this.pluginsController.accountMessageHandlers.get(uniquePluginAccountDomains[0])
+    if (uniquePluginAccountDomains.length === 0 || !handler) {
+      throw new Error(`No handler plugin for account ${address} found.`)
+    }
+
+    return handler
+  }
+
+  getOrigin (address) {
```

Comment:
This function is being called without `this.` prefix throughout the file. A hacker at the DeFi hackathon found the issue and will be pushing the fix upstream!

Reply:
Fix in #90